### PR TITLE
ET-4885 allow to prefix snippets

### PR DESCRIPTION
**Pull request by rustonaut**
Allows specifying two prefixes:

- `embedding.prefix.query` prefix all queries with this before computing their embedding
- `embedding.prefix.snippet` prefix all snippets with this before computing their embedding

works for both local and sagemaker setups

Env variables:

- `XAYN_WEB_API__EMBEDDING__PREFIX__QUERY`
- `XAYN_WEB_API__EMBEDDING__PREFIX__SNIPPET`

### CODE REVIEW

File: web-api/src/embedding.rs [hunk 1]
```diff
@@ -128,19 +161,39 @@ impl Embedder {
...
+            (
+                EmbeddingKind::Content,
+                Prefix {
+                    snippet: content, ..
+                },
+            ) => content,
+        };
+
+        let sequence = if let Some(prefix) = prefix {
+            Cow::Owned(format!("{prefix} {sequence}"))
```
Comment (rustonaut) [inline web-api/src/embedding.rs:180]: Is the concatenation using a space okay?

File: web-api/src/embedding.rs [hunk 2]
```diff
@@ -33,6 +35,16 @@ impl Default for Config {
     }
 }
 
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+#[serde(default)]
+#[cfg_attr(test, serde(deny_unknown_fields))]
+pub(crate) struct Prefix {
+    /// Prefix prepended to search queries when embedding them.
+    pub(crate) query: Option<String>,
```
Comment (acrrd) [inline web-api/src/embedding.rs:43]: I am not sure if we care to distinguish between prefix not given and an empty prefix. The end result will be the same.

**Review by acrrd — APPROVED** [review]

File: web-api/src/embedding.rs (hunk 1 above)
Comment (rustonaut) [reply web-api/src/embedding.rs]: removed

File: web-api/src/embedding.rs (hunk 2 above)
Comment (rustonaut) [reply web-api/src/embedding.rs]: If you by default separate the prefix with a white-space from the rest it makes a difference.

Through we removed the default space separator so now it's fine.

**Pull request merged**